tests/v2/validation/publicapi/clusterroletemplatebindings/crtb_create_steve_test.go

```diff
@@ -0,0 +1,303 @@
+	inputData[userPrincipalName] = "local://" + newUser.ID
+
+	resp, err := client.Steve.SteveType(crtbApiEndPoint).Create(inputData)
+	require.NoError(t, err)
+	assert.Contains(t, (*resp).JSONResp[ID], crtbName)
+	assert.Equal(t, (*resp).JSONResp[userPrincipalName], inputData[userPrincipalName])
+}
+
+func createCrtbAndInputNotExistedField(t *testing.T, client *rancher.Client, clusterID string, newUser *management.User, readJson []byte) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	require.NoError(t, err)
+	crtbName := namegen.RandStringLower(crtbNameWithTenChar)
+
+	inputData := unmarshalCrtbJsonTemplate(t, readJson, clusterID, crtbName, newUser.ID, defaultRoleTemplateName)
+	require.NoError(t, err)
+	inputData["notExisted"] = "fieldNotExisted"
+
+	resp, err := client.Steve.SteveType(crtbApiEndPoint).Create(inputData)
+	require.NoError(t, err)
+	assert.Contains(t, (*resp).JSONResp[ID], crtbName)
+	assert.NotContains(t, (*resp).JSONResp, "notExisted")
+}
+
+func validateErrorAlreadyCrtbExisted(t *testing.T, client *rancher.Client, clusterID string, newUser *management.User, readJson []byte) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	require.NoError(t, err)
+	crtbName := namegen.RandStringLower(crtbNameWithTenChar)
+
+	inputData := unmarshalCrtbJsonTemplate(t, readJson, clusterID, crtbName, newUser.ID, defaultRoleTemplateName)
+	require.NoError(t, err)
+
+	resp, err := client.Steve.SteveType(crtbApiEndPoint).Create(inputData)
+	require.NoError(t, err)
+	assert.Contains(t, (*resp).JSONResp[ID], crtbName)
+
+	_, err = client.Steve.SteveType(crtbApiEndPoint).Create(inputData)
+	require.Error(t, err)
+	require.ErrorContains(t, err, crtbConflictError)
+	require.ErrorContains(t, err, crtbAlreadyExisted)
+}
+
+func validateErrorRoleTemplateNameNotExisted(t *testing.T, client *rancher.Client, clusterID string, newUser *management.User, readJson []byte) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	require.NoError(t, err)
+	crtbName := namegen.RandStringLower(crtbNameWithTenChar)
+
+	inputData := unmarshalCrtbJsonTemplate(t, readJson, clusterID, crtbName, newUser.ID, defaultRoleTemplateName)
+	require.NoError(t, err)
+	inputData[roleTemplateName] = "roleTemplateNameNotExisted"
+
+	_, err = client.Steve.SteveType(crtbApiEndPoint).Create(inputData)
+	require.Error(t, err)
+	require.ErrorContains(t, err, badRequest)
+}
+
+func validateErrorBadRequestByPassingDifferentClusterAndNameSpace(t *testing.T, client *rancher.Client, clusterID string, newUser *management.User, readJson []byte) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	require.NoError(t, err)
+	crtbName := namegen.RandStringLower(crtbNameWithTenChar)
+
+	inputData := unmarshalCrtbJsonTemplate(t, readJson, clusterID, crtbName, newUser.ID, defaultRoleTemplateName)
+	require.NoError(t, err)
+	inputData[clusterName] = localCluster
```

Comment:
The test name indicates that we are passing different cluster AND namespace. However, we are only setting cluster here and not namespace. Is this intentional? Should the test be renamed or should we set namespace here?

Reply:
Yah, it is intentional. The requirement says clusterName and namespaceNames should be the same, here i m validating the negative case where I m passing clusterName as localCluster and namespace as downstream clusterId So that I am expecting an Error.